src/v0/destinations/am/deleteUsers.js

```diff
@@ -12,37 +12,39 @@ const userDeletionHandler = async (userAttributes, config) => {
     throw new CustomError("api key/secret for deletion not present", 400);
   }
 
-  for (let i = 0; i < userAttributes.length; i += 1) {
-    const uId = userAttributes[i].userId;
-    if (!uId) {
-      throw new CustomError("User id for deletion not present", 400);
-    }
-    const data = { user_ids: [uId], requester: "RudderStack" };
-    const requestOptions = {
-      method: "post",
-      url: "https://amplitude.com/api/2/deletions/users",
-      headers: {
-        "Content-Type": "application/json",
-        Authorization: `Basic ${btoa(`${apiKey}:${apiSecret}`)}`
-      },
-      data
-    };
-    const resp = await httpSend(requestOptions);
-    if (!resp || !resp.response) {
-      throw new CustomError("Could not get response", 500);
-    }
-    if (
-      resp &&
-      resp.response &&
-      resp.response.response &&
-      resp.response.response.status !== 200 // am sends 400 for any bad request or even if user id is not found. The text is also "Bad Request" so not handling user not found case
-    ) {
-      throw new CustomError(
-        resp.response.response.statusText || "Error while deleting user",
-        resp.response.response.status
-      );
-    }
-  }
+  await Promise.all(
+    userAttributes.map(async ua => {
+      const uId = ua.userId;
+      if (!uId) {
+        throw new CustomError("User id for deletion not present", 400);
+      }
+      const data = { user_ids: [uId], requester: "RudderStack" };
+      const requestOptions = {
+        method: "post",
+        url: "https://amplitude.com/api/2/deletions/users",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Basic ${btoa(`${apiKey}:${apiSecret}`)}`
+        },
+        data
+      };
+      const resp = await httpSend(requestOptions);
+      if (!resp || !resp.response) {
+        throw new CustomError("Could not get response", 500);
+      }
+      if (
+        resp &&
+        resp.response &&
+        resp.response.response &&
+        resp.response.response.status !== 200 // am sends 400 for any bad request or even if user id is not found. The text is also "Bad Request" so not handling user not found case
+      ) {
+        throw new CustomError(
+          resp.response.response.statusText || "Error while deleting user",
+          resp.response.response.status
+        );
+      }
+    })
```

Comment:
We can avoid these complex logic's with `processAxiosResponse` and handle the retryable and throttled statuses